h2o-r/h2o-package/R/word2vec.R

```diff
@@ -1,108 +1,53 @@
                  epochs = epochs)
 
-  res <- .h2o.__remoteSend(.h2o.__W2V, .params = params)
-  .h2o.__waitOnJob(res$job)
-  dest_key <- .h2o.__remoteSend(paste0(.h2o.__JOBS, "/", res$job))$jobs[[1L]]$dest$name
-  w2vmodel <- h2o.getModel(dest_key)
-  new("H2OW2V", h2o = trainingH2OFrame@conn, key = dest_key, train.data=trainingH2OFrame)
+  .h2o.modelJob("word2vec", params, h2oRestApiVersion = 3)
 }
 
 ##
-## Find Synonyms Using an H2OW2V object
+## Find synonyms using a word2vec model.
 ##
-##  @param word2vec: An H2OW2V model.
-##  @param target: A single word, or a vector of words.
+##  @param word2vec: A word2vec model.
+##  @param target: Single word to find synonyms for.
 ##  @param count: The top `count` synonyms will be returned.
 ##
-h2o.synonym<-
-function(word2vec, target, count) {
-  if (!is(word2vec, "H2OW2V")) stop("`word2vec` must be an H2O word2vec object. See h2o.word2vec")
-  if (missing(target)) stop("`target` must be specified")
-  if (!is.character(target)) stop("`target` must be character")
-  if (missing(count)) stop("`count` must be specified")
-  if (!is.numeric(count)) stop("`count` must be numeric")
-
-  params <- list(key = h2o.getId(word2vec), target=target, cnt=count)
-  if (length(target) == 1L) {
-    res <- .h2o.__remoteSend(word2vec@conn, .h2o.__SYNONYMS, .params = params)
-    fr <- data.frame(synonyms = res$synonyms, cosine.similarity = res$cos_sim)
-    fr[with(fr, order(-cosine.similarity)),]
-  } else {
-    .NotYetImplemented()
-#    vecs <- lapply(target, h2o.transform, word2vec)
-#    vec <- colSums(as.data.frame(vecs))
-#    params$vec <- vec
-#    res <- .h2o.__remoteSend(data@conn, .h2o.__SYNONYMS, params)
-#    return(h2o.getFrame(res$key))
-  }
-}
+h2o.findSynonyms <- function(word2vec, target, count = 20) {
+    if (missing(target)) stop("`target` must be specified")
+    if (!is.character(target)) stop("`target` must be character")
+    if (missing(count)) stop("`count` must be specified")
+    if (!is.numeric(count)) stop("`count` must be numeric")
 
-#'
-#' Transform A Word to A Vec Using Word2Vec
-#'
-#' Use a pre-existing word2vec object to transform a target word
-#' into a numeric vector.
-#setMethod("h2o.transform", "H2OW2V", function(word2vec, target) {
-#  if (!is(word2vec, "H2OW2V")) stop("`word2vec` must be an H2O word2vec object. See h2o.word2vecs")
-#  if (missing(target)) stop("`target` must be specified")
-#  if (!is.character(target)) stop("`target` must be character")
-#  if (length(target) > 1) stop("`target` must be a single word")
-#
-#  params <- params <- c(data = word2vec@word2vec:"id", target = target, word2vec@params)
-#  res <- .h2o.__remoteSend(data@conn, .h2o.__TRANSFORM, params)
-#  res$vec
-#})
+    res <- .h2o.__remoteSend(method="GET", "Word2VecSynonyms", model = word2vec@model_id,
+    target = target, count = count)
+    fr <- data.frame(synonyms = res$words, scores = res$scores)
+    fr[with(fr, order(res$scores, decreasing = TRUE)),]
```

Comment:
If you use `with` you don't have to refer to column by `$`. Also instead of `res$scores` better would be to use `fr$scores`, as we sort `fr`. Putting `res$` there is currently not a problem but is error prone.
AFAIU this should works well here: `fr[with(fr, order(scores, decreasing = TRUE)),]`